processing/src/main/java/org/apache/druid/segment/join/PossiblyNullDimensionSelector.java

```diff
@@ -138,7 +138,8 @@ public int lookupId(@Nullable String name)
       // id 0 is always null for this selector impl.
       return 0;
     } else {
-      return baseSelector.idLookup().lookupId(name) + nullAdjustment;
+      IdLookup idLookup = baseSelector.idLookup();
+      return (idLookup == null ? 0 : idLookup.lookupId(name)) + nullAdjustment;
```

Comment:
should this return 0 if the `idLookup` does not exist?

Reply:
It should probably fail if `idLooup` = null.

Reply:
A null `idLookup` there should be an error condition, since the caller should check if `idLookup()` returned null before calling `lookupId(name)` on the `PossiblyNullDimensionSelector`

Reply:
Added an assert statement with an explanation of why it should never be null in here